src/vendor/perfherder.js

```diff
@@ -128,60 +129,49 @@ const getFramework = async (combo) => {
   await frameworkCache[comboString];
 };
 
-/*
-return an array of {repo, framework} objects
- */
-const findCombo = expression => toPairs(expression)
-  .map((v, k) => {
-    if (k === 'or' || k === 'and') {
-      return selectFrom(v)
-        .map(findCombo)
-        .flatten()
-        .groupBy(['repo', 'framework'])
-        .map(first);
-    }
 
-    const { repo, framework } = v;
+function simpler(v) {
+  const vv = v.filter(x => x !== true);
+  if (vv.length === 0) return true;
+  if (vv.length === 1) return vv[0];
+  return { and: vv };
+}
 
-    if (exists(framework)) {
-      if (missing(repo)) {
-        Log.error(
-          'expecting {{expression}} to have {framework, repo} paired',
-          { expression },
-        );
-      }
 
-      return [{ repo, framework }];
+/*
+ * Expecting props to be an array of N property names
+ * Return a list of expression tuples; meant to represent disjunctive normal
+ * such that
+ *   output[x][i] contains expression on props[i]
+ *   output[x][N] is expressions on everything else
+ */
+const extract = (expression, props) => toPairs(expression)
+  .map((param, op) => {
+    if (op === 'or') {
+      return selectFrom(param).map(e => extract(e, props)).flatten();
+    }
+    if (op === 'and') {
+      return combos(...param.map(e => extract(e, props)))
+        .map(v => zip(...v).map(simpler));
     }
 
-    if (exists(repo)) {
-      Log.error('expecting {{expression}} to have {framework, repo} paired', {
-        expression,
-      });
+    const lookup = Data.zip(props.map((p, i) => ([p, i])));
+    const inProps = array(props.length + 1).map(() => ([]));
+    if (isString(param)) {
+      const i = coalesce(lookup[param], props.length);
+      inProps[i].push({ [op]: param });
+      return [inProps.map(simpler)];
     }
 
-    return [];
+    toPairs(param).forEach((rhs, lhs) => {
+      const i = coalesce(lookup[lhs], props.length);
+      inProps[i].push({ [op]: { [lhs]: rhs } });
+    });
+
+    return [inProps.map(simpler)];
```

Comment:
The old `findCombo()` was dedicated to extracting the repo and framework.  The new `extract` function partitions an expression by parameter name; doing some "algebra" on an array of arrays (representing disjunctive normal form).